### DISCUSSION TITLE
Overall status, road to beta & beyond?

### DISCUSSION THREAD

**Discussion by tudorluca** [square/workflow-kotlin#270] [Q&A] [answered]
Hi all! I run a small digital agency, focused mostly on mobile, and I've been watching the Workflow project for quite some time now. 

I'm trying to assess if it's finally time to add it to our toolbox, hopefully pairing it nicely with Jetpack Compose and never look back (really tired of fragments & all those shenanigans).

As of [v1.0.0-alpha.9](https://github.com/square/workflow-kotlin/releases/tag/v1.0.0-alpha.9):
- what's the overall status of the project?
- what's missing to go to beta & beyond?
- is the current design of the framework going to play nicely with Jetpack Compose UI?

**Comment by andreiverdes** [comment]
Really interested to hear the roadmap for workflow!

**Comment by rjrjr** [comment, marked as answer]
Sorry for the slow response here.

The current Alpha label means that we're trying very hard to stop messing with the API. Once we're confident that no more changes will be coming, we'll graduate to Beta. With a bit of production use of the Beta API under our belts, we'll drop the Beta label and declare this a 1.0. I expect the Beta to be much shorter than this protracted Alpha has been, but then I also thought we'd have been in Beta months ago.

A big asterisk on all this talk of a stable API are our experimental annotations, especially [@WorklfowUiExperimentalApi](https://github.com/square/workflow-kotlin/blob/main/workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/WorkflowUiExperimentalApi.kt). While the core APIs are very solid, there is another round of potentially API-breaking [changes we'd like to land at the Android UI layer](https://github.com/square/workflow-kotlin/labels/ui), especially around [modals](https://github.com/square/workflow-kotlin/issues/195).

Even there, though, there's only so much we can break. The majority of our Square Point of Sale app family is built around this code, and all new feature development relies on it. When things do change, we have to take care to make migration, not just possible, but very practical.

WRT Compose, we intend to adopt it in our production development as soon as we can. We don't know yet exactly what our Workflow-to-Compose migration story will be, but are confident that the unidirectional muscle memory Workflow has allowed us to build over the last few years will serve us well. 

In the meantime, you'll note that a good bit of Compose integration has already been built, and recently [merged into our main repo](https://github.com/square/workflow-kotlin/tree/main/workflow-ui/compose). That's all very fluid, but demonstrates the feasibility of using Compose and Workflow together, and should continue to provide a preview of our eventual roadmap in that direction.
